pkg/cluster/clusterserviceprincipal.go

```diff
@@ -77,47 +78,64 @@ func (m *manager) createOrUpdateClusterServicePrincipalRBAC(ctx context.Context)
 	return nil
 }
 
-func (m *manager) updateAROSecret(ctx context.Context) error {
+// servicePrincipalUpdated checks whether the CSP has been updated by comparing the cluster doc's
+// ServicePrincipalProfile to the contents of the kube-system/azure-cloud-provider Secret. If the CSP
+// has changed, it returns true along with a new corev1.Secret to use to update the Secret to match
+// what's in the cluster doc.
+func (m *manager) servicePrincipalUpdated(ctx context.Context) (bool, *corev1.Secret, error) {
 	var changed bool
```

Comment:
if possible I'd rather see the code that is in the `if changed` block extracted to its own helper function and used along with a `return` when we need to set `changed=true`. this avoids the use of a tracking boolean and reduces complexity.

Reply:
Yeah, I see what you mean. I've pushed a commit that I believe implements this the way you have in mind.